pkg/api/v1/types.go

```diff
@@ -117,9 +117,31 @@ type VMList struct {
 	Items           []VM            `json:"items"`
 }
 
+type MemoryQoS struct {
+	// Request field defines the amount of memory a VM gets upon starting
+	// The format follows k8s quantity specification
+	Request string `json:"request"`
+
+	// Limit defines the maximum amount of memory a VM can ever get in kiloBytes.
+	// This field depends on qemu capabilities, but can be only 4x as big as
+	// `Request` at the time I am writing this (8/2017)
+	// The format follows k8s quantity specification
+	Limit string `json:"limit"`
+}
+
+type VMConfig struct {
+	MemoryQos *MemoryQoS `json:"memory"`
+}
+
 // VMSpec is a description of a VM. Not to be confused with api.DomainSpec in virt-handler.
 // It is expected that v1.DomainSpec will be merged into this structure.
 type VMSpec struct {
+	// TODO High level VM definitions
+	// It is expected the future VMConfig will be somehow processed
+	// into DomainSpec and both representations stored into VMSpec
+	// so VMConfig and DomainSpec describe the same configuration in different
+	// level of detail
+	VMConfig *VMConfig `json:"config,omitempty"`
```

Comment:
This isn't what we're going for with the VMConfig object.

VMConfig is a cluster concept we're still working through.  The VMConfig may actually container the VMSpec object at some point. Given the current discussions, VMConfig is not something that will be a part of the vm spec.

Unfortunately you're getting a bit of conflicting information about the direction of this patch. I'm sorry about that. The source of the confusion is around the relationship between the the VMSpec and the libvirt domain xml. Your PR has exposed that we are not all in agreement about how to proceed with that relationship moving forward.